src/plugins/orangehrmOpenidAuthenticationPlugin/Service/SocialMediaAuthenticationService.php

```diff
@@ -58,26 +91,104 @@ public function initiateAuthentication(AuthProviderExtraDetails $provider, strin
     }
 
     /**
-     * @param OpenIDConnectClient $oidcClient
-     * @throws OpenIDConnectClientException
+     * @return string
+     */
+    public function getRedirectURL(): string
+    {
+        /** @var UrlGenerator $urlGenerator */
+        $urlGenerator = $this->getContainer()->get(Services::URL_GENERATOR);
+        return $urlGenerator->generate('auth_oidc_login_redirect', [], UrlGenerator::ABSOLUTE_URL);
+    }
+
+    /**
+     * @return string
+     */
+    public function getScope(): string
+    {
+        return self::SCOPE;
+    }
+
+    /**
+     * @param UserCredential $userCredential
+     * @return User[]
+     */
+    private function getSystemUsers(UserCredential $userCredential): array
+    {
+        $userSearchFilterParams = new UserSearchFilterParams();
+        $userSearchFilterParams->setUsername($userCredential->getUsername());
+
+        return $this->getUserDao()->searchSystemUsers($userSearchFilterParams);
+    }
+
+    /**
+     * @param UserCredential $userCredentials
+     *
+     * @return User
+     * @throws AuthenticationException
      */
-    public function handleCallback(OpenIDConnectClient $oidcClient): void
+    public function getUserForAuthenticate(UserCredential $userCredentials): User
     {
-        $oidcClient->authenticate();
+        $users = $this->getSystemUsers($userCredentials);
+        if (empty($users)) {
+            throw AuthenticationException::noUserFound();
```

Comment:
check the possibility of user enumeration attack.
cc @devishke-orange

Reply:
check whether we need to handle this as invalid credentials

Reply:
handled as invalid credentials